Enhancement: Add message notification for data file loading process (#285)
AY2324S2-CS2103-F15-1/tp: docs/DeveloperGuide.md

- Modify the pre-defined set of allowed characters to **capture any character instead**. This allows for flexibility since all special characters will be accepted for the `ADDRESS` field.
- For addresses containing `/` characters, the user will be required to use the `\` character to escape each `/` character similar to the example in Planned Enhancement 1.

> ### 3. Add message notification for data file loading process

Reviewer: Do we still require this @Javiery3889 @Dethada @Arixeyeion if we are no longer planning to support editing of JSON file outside of FINDvisor?

Author: This one is for loading of data right? Like I save my contacts in the JSON then just copy over to a new computer?

Reviewer 2: I think we should have it, useful in the case where the data file is corrupted for some reason, might not be due to manual modification.

Reviewer 3: @KohGuanZeh @YuZizhen @Arixeyeion I agreww with what @Dethada mentioned, would be better if we include it

**Current Implementation**: 

- There is **no message notification** for incorrect data file loading and success data loading.
- For example, when load a data file, the user does not know whether the data file is loaded succesfully into the FINDvisor unless the user check the contacts manually to see if there are any changes.

**Proposed Enhancement**: 

- Add a message for successful data file loading and an error message specifying the type of error encountered while loading data file. This provides a more direct feedback for the user.

> - For example, `Data file is loaded succesfully!` will be shown when the loading is succeeded. `Data file does not exist in the directory.` will be returned if the target datafile is not found in the directory.

Author: Ok! I have made the corresponding changes! Thanks :)

- Modify the format of `DATE` to be `d-M-yyyy` and `DATETIME` to be `d-M-yyyy`T`H:mm` instead.
- This allows FINDvisor to accept both single and double-digits day, month and hour values as valid `DATE` and `DATETIME` values and would not require users to pad these single digit values with a leading zero.

### 5. Notfiy user if data file is invalid
**Current Implementation**: 

- No error message is displayed to the user when this occurs and users are not informed of the data loss.
- If new state changing command is made when the data file is invalid, the data file will be overwritten.

**Proposed Enhancement**: 

> - Display a message in the Command Result Box if the data file cannot be parsed on startup.

Reviewer: ```suggestion
- Display a message in the *Command Result Box* if the data file cannot be parsed on startup.
```
I think it is better to make it italics, given that it is how we will reference our UI components.

> - For example, `Data file (<file_location>) could not be loaded!` will be shown when the data file is invalid, where `file_location` is the location of the current referenced data file.

Reviewer 4: ```suggestion
- For example, `Data file (<file_location>) could not be loaded!` will be shown in the command result box when the data file is invalid, where `file_location` is the location of the current referenced data file, so the user can modify it accordingly.
```

Then delete line 443.

Reviewer: I think this error message is sufficient but can be improved by changing `command result box` to `Command Result Box`. Line 443 should also be removed.

- Modify the format of `DATE` to be `d-M-yyyy` and `DATETIME` to be `d-M-yyyy`T`H:mm` instead.
- This allows FINDvisor to accept both single and double-digits day, month and hour values as valid `DATE` and `DATETIME` values and would not require users to pad these single digit values with a leading zero.

### 3. Notfiy user if data file is invalid
**Current Implementation**: 

- If the data file cannot be parsed due to invalid fields, FINDvisor will discard all data and start with an empty data file at the next run.
- No notification will be displayed to the user when this occurs, which might affect the usability of FINDvisor.

**Proposed Enhancement**: 

> - Notify the user if the data file is invalid on startup.

Reviewer 3: ```suggestion
- Display a message in the Command Result Box if the data file cannot be parsed on startup.
```

- Modify the format of `DATE` to be `d-M-yyyy` and `DATETIME` to be `d-M-yyyy`T`H:mm` instead.
- This allows FINDvisor to accept both single and double-digits day, month and hour values as valid `DATE` and `DATETIME` values and would not require users to pad these single digit values with a leading zero.

### 5. Notfiy user if data file is invalid
**Current Implementation**: 

- If the data file cannot be parsed due to invalid fields, FINDvisor will discard all data and start with an empty data file at the next run.

> - No notification will be displayed to the user when this occurs, which might affect the usability of FINDvisor.

Author: Yepp

**Proposed Enhancement**: 

- Display a message in the Command Result Box if the data file cannot be parsed on startup.

> - For example, `Data file (<file_location>) could not be loaded!` will be shown when the data file is invalid, where `file_location` is the location of the current referenced data file.

Reviewer 2: We should specify what we do with the invalid data file too, either still discard it or move it to somewhere.

- Modify the format of `DATE` to be `d-M-yyyy` and `DATETIME` to be `d-M-yyyy`T`H:mm` instead.
- This allows FINDvisor to accept both single and double-digits day, month and hour values as valid `DATE` and `DATETIME` values and would not require users to pad these single digit values with a leading zero.

### 5. Notfiy user if data file is invalid
**Current Implementation**: 

- If the data file cannot be parsed due to invalid fields, FINDvisor will discard all data and start with an empty data file at the next run.
- No error message is displayed to the user when this occurs and users are not informed of the data loss.

**Proposed Enhancement**: 

- Display a message in the Command Result Box if the data file cannot be parsed on startup.
- For example, `Data file (<file_location>) could not be loaded!` will be shown when the data file is invalid, where `file_location` is the location of the current referenced data file.
- Specific error message about what is the error will be shown at the command result box. The user can check the data file and modify it accordingly.

> - If new state changing command is made when the data file is invalid, the data file will be overwritten.

Author: Just a nit, should we mention modification of data file before overwritten of the data file?
